kotlin/e2e-tests/lib/src/main/kotlin/uk/gov/nationalarchives/lib/IngestUtils.kt

```diff
@@ -0,0 +1,225 @@
+    private val dynamoDbClient: DynamoDbClient,
+    private val sfnClient: SfnClient,
+    private val config: Config,
+    private val files: MutableList<UUID>
+) {
+
+    private val completeStatus = "Asset has been written to custodial copy disk."
+    private val failedStatus = "There has been an error ingesting the asset."
+
+    fun checkForValidationFailureMessages(logGroupArn: String, timeout: Long) {
+        streamLogs(timeout, logGroupArn) { logEvents: List<LiveTailSessionLogEvent>? ->
+            logEvents?.let { events ->
+                val assetIds = events
+                    .flatMap {
+                        try {
+                            listOf(jsonCodec.decodeFromString<ValidationErrorMessage>(it.message!!))
+                        } catch (e: Exception) {
+                            emptyList()
+                        }
+                    }
+                    .map { it.fileId }
+                files.removeAll(assetIds)
+                println(files)
+                files.isEmpty()
+            } ?: false
+        }
+    }
+
+
+    fun checkForIngestStatusMessages(logGroupArn: String, timeout: Long, messageType: String) {
+        val status = if (messageType == "update") failedStatus else completeStatus
+        streamLogs(timeout, logGroupArn) { logEvents: List<LiveTailSessionLogEvent>? ->
+            logEvents?.let { events ->
+                val assetIds = events
+                    .map { jsonCodec.decodeFromString<ExternalNotificationMessage>(it.message!!) }
+                    .filter { it.body.properties.messageType == "preserve.digital.asset.ingest.$messageType" && it.body.parameters.status == status }
+                    .map { it.body.parameters.assetId }
+                files.removeAll(assetIds)
+                files.isEmpty()
+            } ?: false
+        }
+    }
+
+    suspend fun createFiles(
+        numberOfFiles: Int,
+        emptyChecksum: Boolean = false,
+        invalidMetadata: Boolean = false,
+        invalidChecksum: Boolean = false
+    ) {
+        val invalidChecksumValue = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+        files.addAll(List<UUID>(numberOfFiles) { UUID.randomUUID() })
+        files.forEach { println("$it $invalidChecksum $emptyChecksum $invalidMetadata") }
+        coroutineScope {
+            files.map {
+                launch {
+                    val checksum = if (emptyChecksum) {
+                        ""
+                    } else if (invalidChecksum) {
+                        invalidChecksumValue
+                    } else {
+                        hash(it.toString())
+                    }
+                    createFile(it.toString(), ByteStream.fromString(it.toString()))
+                    createFile("${it}.metadata", createMetadataJson(it, checksum, invalidMetadata))
+                }
+            }
+        }
+    }
+
+    suspend fun sendMessages() = coroutineScope {
```

Comment:
Should we add the queue name in this method name?

Reply:
I don't think we need to. I've created two methods, one for sending TDR messages and one for judgment messages and they each have their own queue in the config,